pkg/egressgateway/manager.go

```diff
@@ -81,18 +104,51 @@ type Manager struct {
 }
 
 // NewEgressGatewayManager returns a new Egress Gateway Manager.
-func NewEgressGatewayManager(cacheStatus k8s.CacheStatus, identityAlocator identityCache.IdentityAllocator, installRoutes bool) *Manager {
+func NewEgressGatewayManager(cacheStatus k8s.CacheStatus, identityAllocator identityCache.IdentityAllocator, cfg Config) *Manager {
+	return NewEgressGatewayManagerFromParams(Params{
+		CacheStatus:       cacheStatus,
+		IdentityAllocator: identityAllocator,
+		Config:            cfg,
+	})
+}
+
+type Params struct {
+	cell.In
+
+	Config            Config
+	CacheStatus       k8s.CacheStatus
+	IdentityAllocator identityCache.IdentityAllocator
+
+	// Lifecycle to attach hooks to. Optional.
+	Lifecycle hive.Lifecycle
```

Comment:
Is this optional for testing's sake? Seems cleaner to me to construct a hive and get a lifecycle from it in the test.

Reply:
It's optional so that I don't have to change all testing call sites, is that what you mean? Integrating hive into the tests causes the PR to balloon again, for little direct benefit IMO.

Reply:
that's fine, in that case maybe mention that it's optional for testing, and should be supplied in the usual case

Reply:
> Integrating hive into the tests causes the PR to balloon again, for little direct benefit IMO.

Turns out i was full of crap, it's fine. Reworked to start a hive.

Reply:
I might be missing something, but I see neither a hive nor a non-optional Lifecycle - did you back out of this again?